validphys2/src/validphys/commondataparser.py

```diff
@@ -1,18 +1,319 @@
 """
-This module implements parsers for commondata  and systype files into useful
-datastructures, contained in the :py:mod:`validphys.coredata` module, which are
-not backed by C++ managed memory, and so they can be easily pickled and
-interfaces with common Python libraries.  The integration of these objects into
-the codebase is currently work in progress, and at the moment this module
-serves as a proof of concept.
+This module implements parsers for commondata and its associated metadata and uncertainties files
+into useful structures that can be fed to the main :py:class:`validphys.coredata.CommonData` class.
+
+A CommonData file is completely defined by a name (which defines the folder in which the information is)
+and a variant (which defines which files inside of the folder will be read).
+
+
+In this module a few auxiliary dataclasses that hold special information:
+    - TheoryMeta: contains the necessary information to read the (new style) fktables
+    - KinematicsMeta: containins metadata about the kinematics
+    - ReferenceMeta: literature references for the dataset
+
+The CommonMetaData defines how the CommonData file is to be loaded,
+by modifying the CommonMetaData using one of the loaded Variants one can change the resulting
+:py:class:`validphys.coredata.CommonData` object.
 """
-from operator import attrgetter
+from copy import copy
+from functools import cached_property
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Optional
 
 import pandas as pd
+from reportengine.compat import yaml
+from validobj.custom import Parser
+from validobj import ValidationError, parse_input
 
-from validphys.core import peek_commondata_metadata
+from validphys.utils import parse_yaml_inp
 from validphys.coredata import CommonData
 
+EXT = "pineappl.lz4"
+
+
+# Auxiliary parser for common types or sanity checks
+@Parser
+def ValidPath(path_str: str) -> Path:
+    """Parse strings into paths"""
+    return Path(path_str)
+
+
+@Parser
+def ValidOperation(op_str: str) -> str:
```

Comment:
I think we could do `ValidOPeration = Literal[*list_of_operations]`.

Reply:
I'm afraid I don't know what you mean here. How should I write this in `validobj` language?

Reply:
Literally my comment. With ‘list_of_operations’=OP or whatever.